### PR TITLE
Depreate FileInfoTruncated and FileIntf

### PR DESCRIPTION
**Pull request by calmh**
This is open for discussion... It removes the Truncated-stuff and replaces it with a `FileInfo.Truncate()` that nils out the block list and remembers the actual size.

The upside of this is that we get accurate space reporting in the GUI (with the exception of the puller progress, because it's still block based, but that's minor).

The downside is that we unmarshal more data from the DB when we don't really need it, just to throw it away. I don't think this is that big of a deal - most likely the data is already in memory as we get the whole blob of `[]byte` from leveldb as part of `...Get()` or `...Value()` so it's just that we actually unmarshal it as well.

It also removes a bunch of code and types, which is nice.

What do you think?

### CODE REVIEW

File: Godeps/_workspace/src/github.com/syncthing/protocol/message.go [hunk 1]
```diff
@@ -28,6 +30,9 @@ func (f FileInfo) String() string {
 }
 
 func (f FileInfo) Size() (bytes int64) {
+	if f.ActualSize > 0 {
+		return f.ActualSize
+	}
 	if f.IsDeleted() || f.IsDirectory() {
 		return 128
```
Comment (Zillode) [inline Godeps/_workspace/src/github.com/syncthing/protocol/message.go]: First time I noticed this magic constant. Might need a ref/comment why this is 128?

Comment (calmh) [inline Godeps/_workspace/src/github.com/syncthing/protocol/message.go]: Seemed like a nice round number.

(Seriously though, it just represents a small nonzero value, to indicate that there is some work to be done and cause us to be out of sync when we are "missing" deleted files and so on.)

**Comment by AudriusButkevicius** [comment]
Looks good to me.
Fine to merge subject it passes the integration tests, and doesn't bork the UI.
We should probably scrap the squiggles (~) in that case, and fix progress emitter to send size as well as blocks making it more accurate.

**Comment by calmh** [comment]
This is the wrong approach... I'll be back.

**Pull request closed without merging**